packages/eslint-plugin/tests/rules/prefer-as-const.test.ts

```diff
@@ -25,12 +25,12 @@ ruleTester.run('prefer-as-const', rule, {
     "let foo = 'bar';",
     `
       class foo {
-        bar: 'baz' = 'baz';
+        bar = 'baz';
```

Comment:
Now that the rule explicitly handles class properties, we're going to want to test edge cases for them a bit more exhaustively:

* Getters and setters
* Mismatched literal vs. th
* Arrow function with an `as ...`